src/services/users.ts

```
@@ -0,0 +1,39 @@
+import { Request, Response, NextFunction } from 'express'
+
+export default class Users {
+  constructor() {
+  }
+
+  public async middleware(req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+      if (req.type === 'get_user') {
+        await this.getUser(req, res, next)
+      } else if (req.type === 'post_user') {
+        await this.postUser(req, res, next)
+      } else if (req.type === 'put_user') {
+        await this.putUser(req, res, next)
+      } else if (req.type === 'delete_user') {
+        await this.deleteUser(req, res, next)
+      }
+    } catch (e) {
+      console.error(e)
+      // TODO: add in error reporting for failed requests
+    }
+
+    public async getUser(req: Request, res: Response, next: NextFunction): Promise<void> {
+      // TODO: add DB request
```

I tend to like throwing errors rather than silently doing nothing. 

```
throw new Error("unimplemented")
```

This will make it easy to debug what happened if you hit one of these later.

---

good practice